Sources/Xcodeproj/generate().swift

```diff
@@ -69,11 +71,25 @@ public func generate(
     // references in the project.
     let extraDirs = try findDirectoryReferences(path: srcroot)
 
+    // Find non-source files in the source directories and root that should be added
+    // as a reference to the project.
+    var extraFiles = try findNonSourceFiles(path: srcroot)
+    let sourcesExtraFiles = try findNonSourceFiles(path: srcroot.appending(component: "Sources"), recursively: true)
+    extraFiles.append(contentsOf: sourcesExtraFiles)
+    let testsExtraFiles = try findNonSourceFiles(path: srcroot.appending(component: "Tests"), recursively: true)
+    extraFiles.append(contentsOf: testsExtraFiles)
+
+    // Get the ignored files and exclude them
+    let ignoredFiles = try repositoryProvider.openCheckout(at: srcroot).getIgnoredFiles()
+    extraFiles = extraFiles.filter {
+        return !ignoredFiles.contains($0)
+    }
```

Comment:
I think we can pass a closure to walk function that will avoid iterating the ignored directories, which is a little better for performance. We should also get some tests to prove this PR works.